### PR TITLE
Added README for EV3D4

### PR DESCRIPTION
**Pull request by dwalton76**
This is for issue #239

### CODE REVIEW

File: demo/EV3D4/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# EV3D4
+EV3D4 is designed to look like R2-D2 from Star Wars. There are two options for controlling EV3D4.  The first is to use the IR remote to send commands to the IR sensor, run EV3D4RemoteControl to use this method.  The second means of controlling EV3D4 is via a web browser, run EV3D4WebControl to use this method.
```
Comment (WasabiFan) [inline demo/EV3D4/README.md]: Extra whitespace after some of these sentences

File: demo/EV3D4/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+**Building instructions**: https://www.lego.com/en-us/mindstorms/build-a-robot/ev3d4
```
Comment (WasabiFan) [inline demo/EV3D4/README.md]: At the bottom there's no colon for a similar line; can we normalize?

File: demo/EV3D4/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+### EV3D4RemoteControl
+If you are using EV3D4RemoteControl the only other file you need from demo/EV3D4/ is EV3D4.py. The other files are only needed by EV3D4WebControl.
```
Comment (WasabiFan) [inline demo/EV3D4/README.md]: Can we code-format file paths?

File: demo/EV3D4/README.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+EV3D4RemoteControl creates a child class of ev3dev/helper.py's RemoteControlledTank.
+
+
+### EV3D4WebControl
+EV3D4WebControl creates a child class of ev3dev/helper.py's WebControlledTank. The WebControlledTank class runs a web server that serves the web pages, images, etc but it also services the AJAX calls made via the client.  The user loads the initial web page at which point they choose the "Desktop interface" or the "Mobile Interface".
```
Comment (WasabiFan) [inline demo/EV3D4/README.md]: Spaces here too

File: demo/EV3D4/README.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+Desktop Interface - The user is presented with four arrows for moving forwards, backwards, spinning clockwise or spinning counter-clockwise. Two additional buttons are provided for controlling the medium motor. There are two sliders, one to control the speed of the tank and the other to control the speed of the medium motor.
+
+Mobile Interface - The user is presented with a virtual joystick that is used to control the movements of the robot.  Slide your thumb forward and the robot moves forward, slide it to the right and the robot spins clockwise, etc. The further you move the joystick towards the edge of the circle the faster the robot moves. Buttons and a speed slider for the medium motor are also provided.
+
+Both interfaces have touch support so you can use either Desktop or Mobile from your smartphone. When the user clicks/touches a button some jQuery code will fire off an AJAX call to let the EV3D4 web server know what the user clicked or where the joystick is if using the Mobile Interface. The web server in WebControlledTank services this AJAX call and adjust motors speeds/power accordingly.
```
Comment (WasabiFan) [inline demo/EV3D4/README.md]: Extra "s" at end of line ("motors")

**Comment by dwalton76** [comment]
Thanks, will clean up all of these.

File: demo/EV3D4/README.md [hunk 6]
```diff
@@ -1,24 +1,23 @@
 # EV3D4
-EV3D4 is designed to look like R2-D2 from Star Wars. There are two options for controlling EV3D4.  The first is to use the IR remote to send commands to the IR sensor, run EV3D4RemoteControl to use this method.  The second means of controlling EV3D4 is via a web browser, run EV3D4WebControl to use this method.
+EV3D4 is designed to look like R2-D2 from Star Wars. There are two options for controlling EV3D4. The first is to use the IR remote to send commands to the IR sensor, run EV3D4RemoteControl to use this method. The second means of controlling EV3D4 is via a web browser, run EV3D4WebControl to use this method.
 
 **Building instructions**: https://www.lego.com/en-us/mindstorms/build-a-robot/ev3d4
 
 ### EV3D4RemoteControl
-If you are using EV3D4RemoteControl the only other file you need from demo/EV3D4/ is EV3D4.py. The other files are only needed by EV3D4WebControl.
+If you are using EV3D4RemoteControl the only other file you need from ```demo/EV3D4/``` is EV3D4.py. The other files are only needed by EV3D4WebControl.
```
Comment (WasabiFan) [inline demo/EV3D4/README.md]: Markdown code formatting is just a single backtick.

File: demo/EV3D4/README.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+# EV3D4
+EV3D4 is designed to look like R2-D2 from Star Wars. There are two options for controlling EV3D4. The first is to use the IR remote to send commands to the IR sensor, run EV3D4RemoteControl to use this method. The second means of controlling EV3D4 is via a web browser, run EV3D4WebControl to use this method.
+
+**Building instructions**: https://www.lego.com/en-us/mindstorms/build-a-robot/ev3d4
+
+### EV3D4RemoteControl
+If you are using EV3D4RemoteControl the only other file you need from ```demo/EV3D4/``` is EV3D4.py. The other files are only needed by EV3D4WebControl.
+
+EV3D4RemoteControl creates a child class of ev3dev/helper.py's RemoteControlledTank.
```
Comment (WasabiFan) [inline demo/EV3D4/README.md]: Paths here and below should be code-formatted as well; <kbd>Ctrl</kbd>+<kbd>F</kbd> for `/`

**Comment by WasabiFan** [comment]
Sorry, noticed one more thing but it didn't get posted in the review

**Comment by WasabiFan** [comment]
Also, do we want to wrap lines to 80 chars?

**Pull request merged**